mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -171,7 +171,10 @@ class LinearRegression @Since("1.3.0") (@Since("1.3.0") override val uid: String
    * @group setParam
    */
   @Since("1.6.0")
-  def setSolver(value: String): this.type = set(solver, value)
+  def setSolver(value: String): this.type = {
+    require(Array("auto", "l-bfgs", "normal").contains(value), s"Solver $value was not supported.")
```

Comment:
Yes, good point about the param definition being inherited, I had overlooked that. I think it's ok to use a require here then. I would prefer `Set("auto", ...).contains` but it's a very minor issue.